ReactWindows/ReactNative/Views/Modal/ReactModalHostView.cs

```diff
@@ -0,0 +1,247 @@
+        }
+
+        /// <inheritdoc />                                              
+        protected override AutomationPeer OnCreateAutomationPeer()
+        {
+            return new DynamicAutomationPeer<ReactModalHostView>(this);
+        }
+
+        private void Show()
+        {
+            _contentDialog.ShowAsync().GetResults();
+
+            Window.Current.CoreWindow.KeyDown += CoreWindowOnKeyDown;
+            SystemNavigationManager.GetForCurrentView().BackRequested += OnBackRequested;
+            DisplayInformation.GetForCurrentView().OrientationChanged += OnOrientationChanged;
+            var inputPane = InputPane.GetForCurrentView();
+            inputPane.Hiding += InputPaneOnHidingOrShowing;
+            inputPane.Showing += InputPaneOnHidingOrShowing;
+            this.GetReactContext().GetNativeModule<ExceptionsManagerModule>().BeforeShowDevOptionsDialog += Close;
+        }
+
+        private void CoreWindowOnKeyDown(CoreWindow sender, KeyEventArgs e)
+        {
+            if (e.VirtualKey == VirtualKey.Escape && !e.Handled)
+            {
+                _contentDialog.Hide();
+            }
+        }
+
+        private async void InputPaneOnHidingOrShowing(InputPane sender, InputPaneVisibilityEventArgs args)
+        {
+            // Delay until the input pane has finished hiding or showing
+            await Dispatcher.RunAsync(CoreDispatcherPriority.Low, () =>
+            {
+                SetContentSize();
+            });
+        }
+
+        private void OnBackRequested(object sender, BackRequestedEventArgs e)
+        {
+            e.Handled = true;
+            _contentDialog.Hide();
+        }
+
+        private void OnOrientationChanged(DisplayInformation sender, object args)
+        {
+            SetContentSize();
+        }
+
+        /// <summary>
+        /// Size the dialog content to fill the entire screen
+        /// </summary>
+        private void SetContentSize()
+        {
+            var uiManagerModule = this.GetReactContext().GetNativeModule<UIManagerModule>();
+            var tag = Content.GetTag();
+            var currentCount = ++_resizeCount;
+
+            var contentSize = ApplicationView.GetForCurrentView().VisibleBounds;
+            var inputPane = InputPane.GetForCurrentView();
+
+            // Windows phone has resize issues if you make the modal less then half
+            // the screen size and then rotate the phone
+            if (inputPane.OccludedRect.Height < contentSize.Height / 2)
+                contentSize.Height -= inputPane.OccludedRect.Height;
+
+            uiManagerModule.ActionQueue.Dispatch(() =>
+            {
+                // If multiple events have been dispatched, ignore all but the newest
+                if (currentCount == _resizeCount)
+                {
+                    uiManagerModule.UIImplementation.UpdateRootNodeSize(tag, contentSize.Width, contentSize.Height);
```

Comment:
>uiManagerModule.UIImplementation.UpdateRootNodeSize(tag, contentSize.Width, contentSize.Height); [](start = 20, length = 96)

You should add a small comment pointing to the fact the tag is not really a root view, but it kind of works like one in some regards.